terraform/examples/module_depends_on/README.md

```diff
@@ -0,0 +1,98 @@
+# Example for the `module_depends_on` feature
+
+This example is discussed in more detail in our article
+[xxxxx](xxx).
+
+The code in main.tf defines two resources and two modules where
+one module depends on one of the resources.
+
+```
+module "has-no-dependency" {
+  ....
+}
+
+resource "null_resource" "is-no-dependency" {
+  ....
+}
+
+module "has-external-dependency" {
+  ....
+
+  module_depends_on = [
+    null_resource.is-external-dependency
+  ]
+}
+
+resource "null_resource" "is-external-dependency" {
+  ....
+}
+```
+
+## Running the example
+The example will only create `null` resources as provided by the [null provider]().
+No infrastructure is created anywhere and no costs will be generated.
+Running the example will us the [local backend]() by default to store the terraform state.
+
+### Cloning the repository
+```
+git clone https://github.com/mineiros-io/article-examples.git
+cd article-examples/terraform/examples/module_depends_on
+```
+
+### Initializing Terraform
+Run `terraform init` to initialize the example. The output should look like:
+```
+Initializing modules...
+- has-external-dependency in ../../modules/terraform-null-resources
+- has-no-dependency in ../../modules/terraform-null-resources
+
+Initializing the backend...
+
+Initializing provider plugins...
+- Checking for available provider plugins...
+- Downloading plugin for provider "null" (hashicorp/null) 2.1.2...
+
+Terraform has been successfully initialized!
+
+You may now begin working with Terraform. Try running "terraform plan" to see
+any changes that are required for your infrastructure. All Terraform commands
+should now work.
+
+If you ever set or change modules or backend configuration for Terraform,
+rerun this command to reinitialize your working directory. If you forget, other
+commands will detect it and remind you to do so if necessary.
+```
+
+### Applying the example
+Run `terraform apply -auto-approve` to create the resources.
+You can clearly see in the output how creation of
```

Comment:
We should avoid expressions such as clearly. Let's try to be on point only :)

Reply:
fixed